core/emulate.c

```diff
@@ -81,14 +81,14 @@
 
 /* Emulator ops */
 #define READ_GPR(idx, size) \
-    ctxt->ops->read_gpr(ctxt->vcpu, idx, size)
+    gpr_read(ctxt, idx, size)
 #define WRITE_GPR(idx, value, size) \
```

Comment:
The need for `WRITE_GPR` is very much deminished by this commit - consider removing it entirely.